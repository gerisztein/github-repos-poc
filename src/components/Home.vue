<template lang="pug">
  h1.
    {{ message }}
</template>

<script>
export default {
  name: 'Home',
  data () {
    return {
      message: 'Hello World!'
    }
  }
}
</script>

<style lang="stylus" scoped>
</style>
