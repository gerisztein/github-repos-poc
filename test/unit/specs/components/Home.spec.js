import { shallow, createLocalVue } from 'vue-test-utils'
import Home from '@/components/Home'
import repositoriesList from './_RepositoriesList'
import Vuex from 'vuex'
import VueRouter from 'vue-router';

const localVue = createLocalVue()

localVue.use(Vuex)
localVue.use(VueRouter)

describe('Home.vue', () => {
  let actions
  let getters
  let store
  let wrapper

  const currentQueryParams = {
    langs: undefined,
    order: 'desc',
    page: '1',
    sort: 'stars'
  }

  const currentQuery = {
    order: 'desc',
    page: '1',
    query: 'created:>2018-06-20',
    sort: 'stars'
  }

  const routes = [
    {
      path: '/',
      name: 'Home',
      component: Home
    }
  ]
  const router = new VueRouter({routes})

  beforeEach(() => {
    actions = {
      getRepositoriesList: jest.fn(),
      setQuery: jest.fn(),
      setRepositoriesSortin: jest.fn()
    }

    getters = {
      currentQuery: () => currentQuery,
      currentQueryParams: () => currentQueryParams,
      repositoriesList: () => repositoriesList
    }

    store = new Vuex.Store({
      state: {},
      actions,
      getters
    })

    wrapper = shallow(Home, {
      store,
      localVue,
      router,
      propsData: {
        languages: '',
        order: 'desc',
        page: 1,
        sort: 'stars'
      },
    })
  })

  describe('smoke tests', () => {
    it('should render', () => expect(wrapper.exists()).toBe(true))
  })

  describe('render', () => {
    it('matches snapshot', () => expect(wrapper.html()).toMatchSnapshot())
  })

  describe('vuex store', () => {
    it('repositories list exists', () => expect(store.getters.repositoriesList).toMatchObject(repositoriesList))
  })

})
