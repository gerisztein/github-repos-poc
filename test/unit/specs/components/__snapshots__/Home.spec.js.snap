// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue render matches snapshot 1`] = `
<section class="repositories">
    <header>
        <div class="counter">346,172 repositories found</div>
        <!---->
    </header>
    <div class="repositories__container">
        <!---->
        <section class="list">
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
        </section>
    </div>
</section>
`;
